src/libraries/System.Text.Json/src/System/Text/Json/Document/JsonDocument.cs

```diff
@@ -148,7 +148,29 @@ internal int GetPropertyCount(int index)
 
             CheckExpectedType(JsonTokenType.StartObject, row.TokenType);
 
-            return row.SizeOrLength;
+            int propertyCount = 0;
+            int objectOffset = index + DbRow.Size;
+            int innerDepth = 0;
+            for (; objectOffset < _parsedData.Length; objectOffset += DbRow.Size)
```

Comment:
The issue with this approach is that needs to perform a traversal of the entire object to compute the size. This is already possible with `EnumerateObject()`, we should only include this API if we can guarantee that it's constant-time. 

Were you able to determine what the `row.SizeOrLength` value represents in the case of objects and if so, can be meaningfully used (or changed) so that we can obtain the property count?